config/auth_metadata_service.go

```diff
@@ -0,0 +1,166 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+)
+
+// timeout to wait for metadata service to respond
+const metadataServiceTimeout = 10 * time.Second
+
+const MetadataServiceVersion = "1"
+
+// response expected from the metadata service
+// other fields such as `cluster_id` might be added later
+type serverResponse struct {
+	// URL of the Databricks workspace to connect to
+	Host string `json:"host"`
+
+	// The requested access token.
+	Token oauth2.Token `json:"token"`
+}
+
+// Credentials provider that fetches a token from a locally running HTTP server
+//
+// The credentials provider will perform a GET request to the configured URL.
+//
+// The MUST return 4xx response if the "X-Databricks-Metadata-Version" header
+// is not set or set to a version that the server doesn't support.
+//
+// The server MUST guarantee stable sessions per URL path. That is, if the
+// server returns a token for a Host on a given URL path, it MUST continue to return
+// tokens for the same Host.
+//
+// The server is expected to return a JSON response with the following fields:
+//
+// host: URL of the Databricks host to connect to.
+// token: object with the following fields
+//   access_token: The requested access token.
+//	 token_type: The type of token, which is a "Bearer" access token.
+//	 expires_on: The timespan when the access token expires.
+
+type MetadataServiceCredentials struct{}
+
+func (c MetadataServiceCredentials) Name() string {
+	return "metadata-service"
```

Comment:
please don't rename the name of auth type every other week - it pollutes the stats :) 
<img width="536" alt="image" src="https://user-images.githubusercontent.com/259697/229082992-639ae0b4-b1c7-4fac-8753-34f903c71c7e.png">

Reply:
Isn't that the purpose of a review to align on names and update after a discussion. I don't want to end up in a situation where we need to agree on names even before we start coding.